middleware/TestingHandler.md

```diff
@@ -1,10 +1,16 @@
 # Testing Handler
-A piece of middleware that helps to mock different kinds of responses.
+A piece of middleware that helps to mock different kinds of responses. The goal of this testing handler is to allow a developer to test their application code with random errors injected into their application so that they can see whether the application handles server generated failures. Additionally, a developer can set manual responses to test specific conditions of our client library and the Graph API. 
 
 ## Objectives
-Provide a component, so that application developers can test their application with different kinds of responses (different response codes).
+Provide a configurable testing handler component that, by default, generates random error responses, and can be configured to generate tester specified errors.
 
 ## Requirements
+* MUST support response pass-through for non-intercepted responses.
+* The rate of error responses returned MUST be configurable with a value that represents the percentage of responses intercepted and replaced with a random error response. 
```

Comment:
what should be the default value for this configurable percentage?

Reply:
10%